flink-ml-lib/src/main/java/org/apache/flink/ml/classification/logisticregression/LogisticRegressionModel.java

```diff
@@ -0,0 +1,173 @@
+        DataStream<LogisticRegressionModelData> modelDataStream =
+                LogisticRegressionModelData.getModelDataStream(modelDataTable);
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(),
+                                BasicTypeInfo.DOUBLE_TYPE_INFO,
+                                TypeInformation.of(DenseVector.class)),
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldNames(),
+                                getPredictionCol(),
+                                getRawPredictionCol()));
+        DataStream<Row> predictionResult =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(inputStream),
+                        Collections.singletonMap(broadcastModelKey, modelDataStream),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.map(
+                                    new PredictLabelFunction(broadcastModelKey, getFeaturesCol()),
+                                    outputTypeInfo);
+                        });
+        return new Table[] {tEnv.fromDataStream(predictionResult)};
+    }
+
+    /** A utility function used for prediction. */
+    private static class PredictLabelFunction extends RichMapFunction<Row, Row> {
+
+        private final String broadcastModelKey;
+
+        private final String featuresCol;
+
+        private DenseVector coefficient;
+
+        public PredictLabelFunction(String broadcastModelKey, String featuresCol) {
+            this.broadcastModelKey = broadcastModelKey;
+            this.featuresCol = featuresCol;
+        }
+
+        @Override
+        public Row map(Row dataPoint) {
+            if (coefficient == null) {
+                LogisticRegressionModelData modelData =
+                        (LogisticRegressionModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastModelKey).get(0);
+                coefficient = modelData.coefficient;
+            }
+            DenseVector features = (DenseVector) dataPoint.getField(featuresCol);
+            Tuple2<Double, DenseVector> predictionResult = predictRaw(features, coefficient);
+            return Row.join(dataPoint, Row.of(predictionResult.f0, predictionResult.f1));
+        }
+    }
+
+    /**
+     * The main logic that predicts one input record.
+     *
+     * @param feature The input feature.
+     * @param coefficient The model parameters.
+     * @return The prediction label and the raw probabilities.
+     */
+    private static Tuple2<Double, DenseVector> predictRaw(
+            DenseVector feature, DenseVector coefficient) {
```

Comment:
Does this also works with mulple-classes cases ?

Reply:
It may not work with multi-classes case. In that case, we probably need to refactor this method since the the `coefficient` here is supposed to be a `matrix`. We could do this later.